shared/modules/transaction.utils.js

```diff
@@ -61,3 +81,53 @@ export function txParamsAreDappSuggested(transaction) {
       transaction?.dappSuggestedGasFees?.maxFeePerGas === maxFeePerGas)
   );
 }
+
+/**
+ * Determines the type of the transaction by analyzing the txParams.
+ * This method will return one of the types defined in shared/constants/transactions
+ * It will never return TRANSACTION_TYPE_CANCEL or TRANSACTION_TYPE_RETRY as these
+ * represent specific events that we control from the extension and are added manually
+ * at transaction creation.
+ *
+ * @param {Object} txParams - Parameters for the transaction
+ * @param {EthQuery} query - EthQuery instance
+ * @returns {InferTransactionTypeResult}
+ */
+export async function determineTransactionType(txParams, query) {
+  const { data, to } = txParams;
+  let name;
+  try {
+    name = data && hstInterface.parseTransaction({ data }).name;
+  } catch (error) {
+    log.debug('Failed to parse transaction data.', error, data);
+  }
+
+  const tokenMethodName = [
+    TRANSACTION_TYPES.TOKEN_METHOD_APPROVE,
+    TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER,
+    TRANSACTION_TYPES.TOKEN_METHOD_TRANSFER_FROM,
+  ].find((methodName) => isEqualCaseInsensitive(methodName, name));
+
+  let result;
+  if (data && tokenMethodName) {
+    result = tokenMethodName;
+  } else if (data && !to) {
+    result = TRANSACTION_TYPES.DEPLOY_CONTRACT;
+  }
+
+  let contractCode;
+
+  if (!result) {
+    const {
+      contractCode: resultCode,
+      isContractAddress,
+    } = await readAddressAsContract(query, to);
+
+    contractCode = resultCode;
+    result = isContractAddress
+      ? TRANSACTION_TYPES.CONTRACT_INTERACTION
+      : TRANSACTION_TYPES.SIMPLE_SEND;
+  }
+
+  return { type: result, getCodeResponse: contractCode };
```

Comment:
I know `getCodeResponse` was used in the previous code but it seems like a weird property name, unless it's a function.